pkg/cmd/util_commands.go

```diff
@@ -0,0 +1,97 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+var (
+	ctx9, cancel9 = context.WithCancel(context.Background()) // preemptive: kill subprocess
+	ctx, cancel   = context.WithCancel(ctx9)                 // cooperative: wait for subprocess
+)
+
+func formatRoutes(files []string) []string {
+	routes := []string{}
+	for _, route := range files {
+		// Split route path.
+		a := strings.Split(route, ".")
+
+		// Extract extension.
+		extension := a[len(a)-1]
+
+		// Add file if extension is supported.
+		routes = append(routes, "file:"+route+"?language="+extension)
+	}
+
+	return routes
+}
+
+func confDirectories(properties []string) []string {
+	confDirs := []string{}
+
+	for _, propertiesPath := range properties {
+		confDirs = append(confDirs, path.Dir(propertiesPath))
+	}
+
+	return confDirs
+}
+
+func assembleClasspatchArgValue(properties []string, dependencies []string, routes []string) string {
+	classpathContents := []string{}
+	classpathContents = append(classpathContents, properties...)
+	classpathContents = append(classpathContents, routes...)
+	classpathContents = append(classpathContents, dependencies...)
+	return strings.Join(classpathContents, ":")
+}
+
+// RunLocalIntegration --
+func RunLocalIntegration(properties []string, dependencies []string, routes []string) error {
+	// Create classpath value.
+	classpathValue := assembleClasspatchArgValue(properties, dependencies, routes)
+
+	// Create java command that runs the integration.
+	javaCmd := "java"
+
+	// Create java command arguments.
+	args := make([]string, 0)
+	args = append(args, "-cp")
+	args = append(args, classpathValue)
+	args = append(args, "io.quarkus.runner.GeneratedMain")
+
+	cmd := exec.CommandContext(ctx, javaCmd, args...)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	// cmd.Dir = ctx.Path
+
+	// TODO: for debug purposes, remove when done.
+	fmt.Printf("executing: %s", strings.Join(cmd.Args, " "))
```

Comment:
I think it's ok to keep it. The container on Kube prints this as well.

Reply:
Probably I'd remove the `outputDependencies` call that prints the list of dependencies before it:

```
[nferraro@localhost camel-k]$ kamel local-run examples/routes.js 
/home/nferraro/.m2/repository/org/apache/camel/quarkus/camel-quarkus-timer/1.1.0/camel-quarkus-timer-1.1.0.jar
/home/nferraro/.m2/repository/org/apache/camel/quarkus/camel-quarkus-core/1.1.0/camel-quarkus-core-1.1.0.jar
/home/nferraro/.m2/repository/io/quarkus/quarkus-core/1.8.0.Final/quarkus-core-1.8.0.Final.jar
/home/nferraro/.m2/repository/jakarta/annotation/jakarta.annotation-api/1.3.5/jakarta.annotation-api-1.3.5.jar
/home/nferraro/.m2/repository/jakarta/enterprise/jakarta.enterprise.cdi-api/2.0.2/jakarta.enterprise.cdi-api-2.0.2.jar
/home/nferraro/.m2/repository/jakarta/el/jakarta.el-api/3.0.3/jakarta.el-api-3.0.3.jar
/home/nferraro/.m2/repository/jakarta/interceptor/jakarta.interceptor-api/1.2.5/jakarta.interceptor-api-1.2.5.jar
/home/nferraro/.m2/repository/jakarta/inject/jakarta.inject-api/1.0/jakarta.inject-api-1.0.jar
/home/nferraro/.m2/repository/io/quarkus/quarkus-ide-launcher/1.8.0.Final/quarkus-ide-launcher-1.8.0.Final.jar
/home/nferraro/.m2/repository/io/smallrye/config/smallrye-config/1.8.6/smallrye-config-1.8.6.jar
/home/nferraro/.m2/repository/io/smallrye/common/smallrye-common-annotation/1.3.0/smallrye-common-annotation-1.3.0.jar
/home/nferraro/.m2/repository/io/smallrye/config/smallrye-config-common/1.8.6/smallrye-config-common-1.8.6.jar
/home/nferraro/.m2/repository/io/smallrye/common/smallrye-common-expression/1.1.0/smallrye-common-expression-1.1.0.jar
/home/nferraro/.m2/repository/io/smallrye/common/smallrye-common-function/1.1.0/smallrye-common-function-1.1.0.jar
/home/nferraro/.m2/repository/io/smallrye/common/smallrye-common-constraint/1.1.0/smallrye-common-constraint-1.1.0.jar
/home/nferraro/.m2/repository/org/jboss/logging/jboss-logging/3.3.2.Final/jboss-logging-3.3.2.Final.jar
/home/nferraro/.m2/repository/org/jboss/logmanager/jboss-logmanager-embedded/1.0.4/jboss-logmanager-embedded-1.0.4.jar
/home/nferraro/.m2/repository/org/jboss/logging/jboss-logging-annotations/2.1.0.Final/jboss-logging-annotations-2.1.0.Final.jar
/home/nferraro/.m2/repository/org/jboss/threads/jboss-threads/3.1.1.Final/jboss-threads-3.1.1.Final.jar
/home/nferraro/.m2/repository/org/slf4j/slf4j-api/1.7.30/slf4j-api-1.7.30.jar
/home/nferraro/.m2/repository/org/jboss/slf4j/slf4j-jboss-logging/1.2.0.Final/slf4j-jboss-logging-1.2.0.Final.jar
/home/nferraro/.m2/repository/org/graalvm/sdk/graal-sdk/20.2.0/graal-sdk-20.2.0.jar
/home/nferraro/.m2/repository/org/wildfly/common/wildfly-common/1.5.4.Final-format-001/wildfly-common-1.5.4.Final-format-001.jar
/home/nferraro/.m2/repository/io/quarkus/quarkus-bootstrap-runner/1.8.0.Final/quarkus-bootstrap-runner-1.8.0.Final.jar
/home/nferraro/.m2/repository/io/quarkus/quarkus-arc/1.8.0.Final/quarkus-arc-1.8.0.Final.jar
/home/nferraro/.m2/repository/io/quarkus/arc/arc/1.8.0.Final/arc-1.8.0.Final.jar
/home/nferraro/.m2/repository/jakarta/transaction/jakarta.transaction-api/1.3.3/jakarta.transaction-api-1.3.3.jar
/home/nferraro/.m2/repository/org/eclipse/microprofile/context-propagation/microprofile-context-propagation-api/1.0.1/microprofile-context-propagation-api-1.0.1.jar
/home/nferraro/.m2/repository/org/apache/camel/quarkus/camel-quarkus-support-common/1.1.0/camel-quarkus-support-common-1.1.0.jar
/home/nferraro/.m2/repository/io/quarkus/quarkus-development-mode-spi/1.8.0.Final/quarkus-development-mode-spi-1.8.0.Final.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-core-engine/3.5.0/camel-core-engine-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-api/3.5.0/camel-api-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-management-api/3.5.0/camel-management-api-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-util/3.5.0/camel-util-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-base/3.5.0/camel-base-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-core-languages/3.5.0/camel-core-languages-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-endpointdsl/3.5.0/camel-endpointdsl-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-componentdsl/3.5.0/camel-componentdsl-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-core-catalog/3.5.0/camel-core-catalog-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-tooling-model/3.5.0/camel-tooling-model-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-util-json/3.5.0/camel-util-json-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-microprofile-config/3.5.0/camel-microprofile-config-3.5.0.jar
/home/nferraro/.m2/repository/org/eclipse/microprofile/config/microprofile-config-api/1.4/microprofile-config-api-1.4.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-timer/3.5.0/camel-timer-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-support/3.5.0/camel-support-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/quarkus/camel-quarkus-log/1.1.0/camel-quarkus-log-1.1.0.jar
/home/nferraro/.m2/repository/org/apache/camel/camel-log/3.5.0/camel-log-3.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/k/camel-k-quarkus-loader-js/1.5.0/camel-k-quarkus-loader-js-1.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/k/camel-k-quarkus-core/1.5.0/camel-k-quarkus-core-1.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/k/camel-k-runtime-core/1.5.0/camel-k-runtime-core-1.5.0.jar
/home/nferraro/.m2/repository/org/apache/camel/quarkus/camel-quarkus-endpointdsl/1.1.0/camel-quarkus-endpointdsl-1.1.0.jar
/home/nferraro/.m2/repository/org/apache/camel/k/camel-k-loader-js/1.5.0/camel-k-loader-js-1.5.0.jar
/home/nferraro/.m2/repository/org/graalvm/js/js/20.2.0/js-20.2.0.jar
/home/nferraro/.m2/repository/org/graalvm/regex/regex/20.2.0/regex-20.2.0.jar
/home/nferraro/.m2/repository/org/graalvm/truffle/truffle-api/20.2.0/truffle-api-20.2.0.jar
/home/nferraro/.m2/repository/org/ow2/asm/asm/8.0.1/asm-8.0.1.jar
/home/nferraro/.m2/repository/org/ow2/asm/asm-tree/8.0.1/asm-tree-8.0.1.jar
/home/nferraro/.m2/repository/org/ow2/asm/asm-analysis/8.0.1/asm-analysis-8.0.1.jar
/home/nferraro/.m2/repository/org/ow2/asm/asm-commons/7.1/asm-commons-7.1.jar
/home/nferraro/.m2/repository/org/ow2/asm/asm-util/7.1/asm-util-7.1.jar
/home/nferraro/.m2/repository/com/ibm/icu/icu4j/67.1/icu4j-67.1.jar
/tmp/maven-877782304/target/camel-k-integration-1.3.0-SNAPSHOT-runner.jar
executing: java -cp examples/routes.js:/home/nferraro/.m2/repository/org/apache
```

Reply:
Done